pkg/liquidity-mining/test/BaseGaugeFactory.test.ts

```diff
@@ -0,0 +1,52 @@
+import { Contract } from 'ethers';
+
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { expect } from 'chai';
+import { ANY_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+import { fp } from '@balancer-labs/v2-helpers/src/numbers';
+
+describe('BaseGaugeFactory', () => {
+  let gaugeImplementation: Contract;
+  let gaugeFactory: Contract;
+
+  sharedBeforeEach('deploy gauge factory', async () => {
+    gaugeImplementation = await deploy('MockLiquidityGauge');
+    gaugeFactory = await deploy('MockLiquidityGaugeFactory', { args: [gaugeImplementation.address] });
+  });
+
+  describe('getGaugeImplementation', () => {
+    it('returns the implementation given in the constructor', async () => {
+      expect(await gaugeFactory.getGaugeImplementation()).to.be.eq(gaugeImplementation.address);
+    });
+  });
+
+  describe('create', () => {
+    it('emits an event', async () => {
+      const tx = await gaugeFactory.create(ANY_ADDRESS, fp(1)); // Weight cap can be anything; it's not under test.
```

Comment:
`create` doesn't need to be on the interface of the gauge factory. But this is fine for now.